### PR TITLE
Update pfs start logic and pfs start api telemetry

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/_service/entry.py [hunk 1]
```diff
@@ -51,24 +51,26 @@ def add_show_status_action(subparsers):
...
+
+    if port:
+        dump_port_to_config(port)
+        validate_port(port, args.force)
+    else:
+        port = get_port_from_config(create_if_not_exists=True)
+        validate_port(port, args.force)
     # Set host to localhost, only allow request from localhost.
     app.logger.info(f"Start Prompt Flow Service on http://localhost:{port}, version: {get_promptflow_sdk_version()}")
     waitress.serve(app, host="127.0.0.1", port=port)
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_service/entry.py:76]: This is a blocking api, so our monitor operation never ends? how we know the latency then ?

Comment (YingChen1996) [reply src/promptflow/promptflow/_sdk/_service/entry.py:76]: what' the latency here mean?

Comment (wangchao1230) [reply src/promptflow/promptflow/_sdk/_service/entry.py:76]: The time used to start the service

Comment (YingChen1996) [reply src/promptflow/promptflow/_sdk/_service/entry.py:76]: update pfs start telemetry behavior according to discussion

File: src/promptflow/promptflow/_sdk/_service/utils/utils.py [hunk 2]
```diff
@@ -100,6 +104,19 @@ def make_response_no_content():
     return make_response("", 204)
 
 
+def check_pfs_service_status(pfs_port) -> bool:
+    """Check if pfs service is running."""
+    try:
+        response = requests.get("http://localhost:{}/heartbeat".format(pfs_port))
+        if response.status_code == 200:
+            logger.info(f"Pfs service is already running on port {pfs_port}.")
```
Comment (zhengfeiwang) [inline src/promptflow/promptflow/_sdk/_service/utils/utils.py:112]: not sure on this level, maybe debug is better? @wangchao1230

Comment (YingChen1996) [reply src/promptflow/promptflow/_sdk/_service/utils/utils.py:112]: changed to debug level

**Review by zhengfeiwang — DISMISSED** [review]

File: src/promptflow/promptflow/_sdk/_service/entry.py [hunk 3]
```diff
@@ -106,8 +132,11 @@ def entry(command_args):
     activity_name = _get_cli_activity_name(cli=parser.prog, args=args)
     logger = get_telemetry_logger()
 
-    with log_activity(logger, activity_name, activity_type=ActivityType.INTERNALCALL):
-        run_command(args)
+    if args.action == "start":
+        start_service(args, logger, activity_name)
+    else:
+        with log_activity(logger, activity_name, activity_type=ActivityType.INTERNALCALL):
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_service/entry.py:138]: Public

File: src/promptflow/promptflow/_sdk/_service/entry.py [hunk 4]
```diff
@@ -135,7 +136,7 @@ def entry(command_args):
     if args.action == "start":
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_service/entry.py:136]: why we need to use the with pattern? 
I orginally think we can do custom two log function call for action start/complete?

Comment (YingChen1996) [reply src/promptflow/promptflow/_sdk/_service/entry.py]: updated

File: src/promptflow/promptflow/_sdk/_service/entry.py [hunk 5]
```diff
@@ -53,25 +68,49 @@ def add_show_status_action(subparsers):
...
+        "-m",
+        "waitress",
+        "--host",
+        "127.0.0.1",
+        f"--port={port}",
+        "--call",
+        "promptflow._sdk._service.entry:get_app",
+    ]
+    if args.synchronous:
+        subprocess.call(cmd)
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_service/entry.py:100]: shall we set the environment variable PF_NO_INTERACTIVE_LOGIN here?

Comment (YingChen1996) [reply src/promptflow/promptflow/_sdk/_service/entry.py:100]: copy from service entry: https://github.com/microsoft/promptflow/blob/b74b863c1910bb5bc156fbc40018db12c2516838/src/promptflow/promptflow/_sdk/_service/entry.py#L86

**Review by wangchao1230 — APPROVED** [review]

**Review by brynn-code — APPROVED** [review]

**Pull request merged**